content/blog/eks-oidc/index.md

```diff
@@ -0,0 +1,128 @@
+---
+date: "2020-05-08"
+title: "Access Control for Pods on AWS EKS"
+authors: ["mike-metral"]
+tags: ["EKS", "RBAC", "Kubernetes"]
+meta_desc: "AWS EKS clusters can use IAM roles and policies for Pods to assign fine-grained access control of AWS services."
+meta_image: cluster.svg
+---
+
+AWS [EKS][aws-eks] clusters can use [IAM][aws-iam] roles and policies for Pods
+to assign fine-grained access control of AWS services. The AWS IAM entities map into Kubernetes
+[RBAC][k8s-rbac] to configure the permissions of Pods that work with AWS
+services.
+
+Together, AWS IAM and Kubernetes RBAC enable least-privileged access for your
+apps, scoped to the appropriate policies and user requirements.
+
+[k8s-rbac]: https://kubernetes.io/docs/reference/access-authn-authz/rbac/
+
+<!--more-->
+
+## Overview
+
+* [Pod Access Control](#pod-access-control)
+* [Create an OIDC provider](#create-an-oidc-provider)
+* [Create IAM for a S3 app](#create-iam-for-a-s3-app)
+* [Deploy a S3 app](#deploy-a-s3-app)
+* [Wrap-Up](#wrap-up)
+* [Next Steps](#next-steps)
+
+## Pod Access Control
+
+AWS EKS supports using IAM entities in a Pod [Service Account][k8s-sa] by
+leveraging an [OIDC provider][aws-oidc] connected to the Kubernetes cluster.
+
+Continuing with the example from the [AWS blog post][aws-pod-iam], when an [S3 app written in Go][s3-app] pushes
+an object to a bucket with the AWS SDK, it will need write access to S3.
+
+When a Pod is launched with a particular Service Account, the OIDC provider works
+with Kubernetes to verify the Pod's identity, and in turn collaborates with the
+[AWS Secure Token Service (STS)][aws-sts] to grant the Pod temporary
+credentials to use with the IAM role.
+
+## Create an OIDC provider
+
+Creating an OIDC provider is as simple as toggling the `createOidcProvider` option
+in the definition of your EKS cluster.
+
+When enabled, the OIDC provider will be created and associated with the
+cluster's OIDC provider URL.
+
+![Create an EKS Cluster and OIDC provider](cluster.svg)
+
+## Create IAM for a S3 app
+
+We'll use the OIDC provider URL and Amazon Resource Name (ARN) to compose the
+[AssumeRoleWithWebIdentity][aws-assume-role-web], and S3 IAM policies that will
+be attached to a new S3 IAM role.
+
+After the role is configured, a Service Account for the S3 Pod will be
+created, and annotated with the ARN of the S3 role to bind the two together.
+
+![Create the AWS IAM and Kubernetes Service Account for the Pod](iam.svg)
```

Comment:
The images were used to break up text-heavy content, and because they render quite nicely as SVGs, so it creates good visuals in between text.

Are you saying that instead of capturing the code samples they be in codeblocks for copy/paste?

Reply:
Yeah, that's right. I don't have a strong opinion on it, but that's the feedback I've received on my own posts.

Reply:
cc @spara Thoughts on codeblocks vs SVGs?

Reply:
checking how it renders

Reply:
I think we should stick to code blocks for copy-paste reasons and readability. While the svgs are clear, the text renders at different sizes, which makes it difficult to read on a mobile device:

![Screen Shot 2020-05-19 at 2 36 31 PM](https://user-images.githubusercontent.com/638672/82370444-2c1b5680-99de-11ea-9acc-aad79d0d7904.png)